wp-content/plugins/tapestry/controller/class.tapestry-controller.php

```diff
@@ -3,4 +3,83 @@
  * Add/update/retrieve Tapestry post and its child nodes
  * 
  */
-class TapestryController {}
\ No newline at end of file
+class TapestryController {
+    /**
+     * Retrieve a Tapestry post
+     * 
+     * @param  Number @postId The Tapestry postId
+     * @return Object Tapestry
+     */
+    public function getTapestry($postId = null) {
+        // TODO: Use $this->postId that's passed in the constructor
+        // TODO: uncomment the two lines below for error handling
+        // after the PR for saving tapestry is merged.
+        /*
+        if (!$this->_isValidPostId($postId)) {
+            return $this->throwsError('INVALID_POST_ID');
+        }
+        */
+        $tapestry = $this->_getTapestryById($postId);
+        return $tapestry;
+    }
+
+    private function _getTapestryById($postId) {
+        $tapestry = get_post_meta($postId, 'tapestry', true);
+
+        $metadatas = array_map(function($nodeId) {
+            return get_metadata_by_mid('post', $nodeId);
+        }, $tapestry->nodes);
+
+        $nodeDatas = array_map(function($metadata) {
+            $postId = $metadata->meta_value->post_id;
+            $nodeData = get_post_meta($postId, 'tapestry_node_data', true);
+            return $this->_updateNodeData($nodeData, $metadata);
+        }, $metadatas);
+
+        $tapestry->nodes = $nodeDatas;
+
+        // TODO: delete the below when being able to create tapestry from scratch
+        $tapestry->links = $this->_getNewLinks($tapestry->links, $nodeDatas);
+        return $tapestry;
+    }
+
+    private function _updateNodeData($nodeData, $metadata) {
```

Comment:
Please use a different word than `update` because "update" could imply something is getting updated in the db. Maybe `_formNodeData`